tests/CMakeLists.txt

```diff
@@ -204,11 +131,14 @@ set_target_properties(mock_process
   RUNTIME_OUTPUT_DIRECTORY_DEBUG "${CMAKE_BINARY_DIR}/bin/mocks"
   RUNTIME_OUTPUT_DIRECTORY_RELEASE "${CMAKE_BINARY_DIR}/bin/mocks")
 
-add_executable(apparmor_parser
-  mock_apparmor_parser.cpp)
+add_dependencies(multipass_tests mock_process)
+
+target_include_directories(multipass_tests
+  BEFORE
+    PRIVATE ${CMAKE_SOURCE_DIR}/src/platform/backends/shared/linux
+)
+add_subdirectory(linux)
+add_subdirectory(qemu)
+include(libvirt/CMakeLists.txt) # with add_subdirectory it's not possible to target_link_libraries
```

Comment:
Because you can only `target_link_libraries` targets built in the same directory. A CMake limitation (by design, I believe).

Reply:
OK, I just tried it and I see what you mean. For the record, [this](https://paste.ubuntu.com/p/wmb3vGKTcx/) would work, but it would reduce the encapsulation you are going for here. Yeah fine, I guess this is not worth the trouble of declaring an `ADDITIONAL_LIBS` var and filling it in the subdirs...